account/manager.go

```diff
@@ -306,7 +312,11 @@ func (m *Manager) InitAccount(ctx context.Context, value btcutil.Amount,
 
 	// With our key obtained, we'll reserve an account with our auctioneer,
 	// who will provide us with their base key and our initial per-batch
-	// key.
+	// key. We'll make sure to acquire the reservation lock throughout the
+	// account funding process to ensure we use the same reservation, as
+	// only one can be active per trader LSAT.
+	m.reservationMtx.Lock()
```

Comment:
Why not lock above, for the entire method? Otherwise assuming we have 1 BTC, and two requests from in for 1 BTC accounts, then one of them will get stuck on coin selection, similar to what happened with Reza.